server/apps/immich/src/modules/schedule-tasks/schedule-tasks.module.ts

```diff
@@ -0,0 +1,29 @@
+import { BullModule } from '@nestjs/bull';
+import { Module } from '@nestjs/common';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { AssetEntity } from '@app/database/entities/asset.entity';
+import { ScheduleTasksService } from './schedule-tasks.service';
+
+@Module({
+  imports: [
+    TypeOrmModule.forFeature([AssetEntity]),
+    BullModule.registerQueue({
+      name: 'thumbnail-generator-queue',
+      defaultJobOptions: {
+        attempts: 3,
+        removeOnComplete: true,
+        removeOnFail: false,
```

Comment:
It will sit in the failed job section. So the mechanism of retrying the job is on us, whether to listening to the failed event and put it back to the queue of have a cronjob to do that (we are using cronjob for two most important jobs - the Wepb generator and video encoding, will be thumbnail generator later on as well)

Reply:
Sounds good, just don't want failed jobs to get queued up in the background and fill the queues up causing memory issues